src/common/variables.c

```diff
@@ -39,59 +39,23 @@ typedef struct dt_variables_data_t
   guint sequence;
 } dt_variables_data_t;
 
-gchar *_string_get_first_variable(gchar *string, gchar *variable)
-{
-  if(g_strrstr(string, "$("))
-  {
-    gchar *pend = string + strlen(string);
-    gchar *p, *e;
-    p = e = string;
-    while(p < pend && e < pend)
-    {
-      while(*p != '$' && *(p + 1) != '(' && p < pend) p++;
-      if(*p == '$' && *(p + 1) == '(')
-      {
-        e = p;
-        while(*e != ')' && e < pend) e++;
-        if(e < pend && *e == ')')
-        {
-          strncpy(variable, p, e - p + 1);
-          variable[e - p + 1] = '\0';
-          return p + 1;
-        }
-        else
-          return NULL;
-      }
-      p++;
-    }
-    return p + 1;
-  }
-  return NULL;
-}
-
 gchar *_string_get_next_variable(gchar *string, gchar *variable)
 {
-  gchar *pend = string + strlen(string);
-  gchar *p, *e;
-  p = e = string;
-  while(p < pend && e < pend)
-  {
-    while(!(*p == '$' && *(p + 1) == '(') && p <= pend) p++;
-    if(*p == '$' && *(p + 1) == '(')
-    {
-      e = p;
-      while(*e != ')' && e < pend) e++;
-      if(e < pend && *e == ')')
-      {
-        strncpy(variable, p, e - p + 1);
-        variable[e - p + 1] = '\0';
-        return p + 1;
-      }
-      else
-        return NULL;
-    }
-  }
-  return NULL;
+	gchar *end = NULL;
+	gchar *start = g_strstr_len(string, strlen(string), "$(");
+	if(start)
+	{
+		end = g_strstr_len(start, strlen(start), ")");
+		if(end)
+		{
+			const int length = end - start + 1;
+			strncpy(variable, start, length);
```

Comment:
Please use `g_strlcpy()`, it makes sure that there is no overflow, and the string is zero-terminated in all cases.